### PR TITLE
Migrate runners to a runners pkg

### PR DESCRIPTION
**Pull request by mkcp**
This completes the code components of the seekers migration. Here we split the combined `op` package into `runner` and `op`.  This created a circular dependency, because Op accepted `Runner` as a param. Instead, we now accept the basic data types and collections at the function boundary rather than calling the methods on the calling object. You can see in the commit messages that there's a couple other fixes to the Results rendering.

Alongside with #147 and #149 , we introduce some breaking changes to the form of Results.json. The "runner" field is now effectively called "params" and renders more consistently across different `Runner` (prev. `seeker`) types. It will be omitted if a runner does does not take any params.

### CODE REVIEW

**Comment by mkcp** [comment]
This includes the changes in #149 so the diff against the merge target, main, contains more than just the new commits. The diff should update once #149 is merged.

File: changelog/150.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:breaking
```
Comment (nwchandler) [inline changelog/150.txt:1]: I wonder if we should also include an improvement note for this PR. There was also a lot done to improve the extensibility of the code going forward. I know it won't directly impact users, but it may warrant a blurb in the 0.4.0 changelog.

File: product/consul.go [hunk 2]
```diff
@@ -7,9 +7,9 @@ import (
 	"github.com/hashicorp/go-hclog"
 
 	"github.com/hashicorp/hcdiag/client"
-	"github.com/hashicorp/hcdiag/op"
-	s "github.com/hashicorp/hcdiag/op"
-	logs "github.com/hashicorp/hcdiag/op/log"
+	"github.com/hashicorp/hcdiag/runner"
+	s "github.com/hashicorp/hcdiag/runner"
```
Comment (nwchandler) [inline product/consul.go:11]: I know this was not part of the PR, but I see that we're importing `runner` directly as well as aliased to `s`. Could we get rid of the alias? It looks like it's only used on line 40.

File: runner/runner.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			newRunners = append(newRunners, r)
+		}
+	}
+	return newRunners, nil
+}
+
+// Params takes a Runner and returns a map of its public fields
+func Params(r Runner) map[string]interface{} {
+	var inInterface map[string]interface{}
+	inrec, _ := json.Marshal(&r)
```
Comment (nwchandler) [inline runner/runner.go:75]: Obligatory "Should we handle this error?" 😄

**Review by nwchandler — APPROVED** [review]
Looks good. I included a few small comments for consideration.

File: changelog/150.txt (hunk 1 above)
Comment (mkcp) [reply changelog/150.txt:1]: True - the changes ought to make contributing runners quite a bit easier!

File: product/consul.go (hunk 2 above)
Comment (mkcp) [reply product/consul.go:11]: There were a few places seeker was aliased - missed this one in the refactor, thanks!

File: runner/runner.go (hunk 3 above)
Comment (mkcp) [reply runner/runner.go]: Ah, got caught copy/pasting 😅
